src/channel.rs

```diff
@@ -1,25 +1,27 @@
 //! ADC input channels
 use crate::{ic, Ads1x1x, BitFlags as BF, Config};
 
-mod private {
-    pub trait Sealed {}
-}
+use private::ChannelSelection;
 
 /// Marker type for an ADC input channel.
-pub trait ChannelId<T>: private::Sealed {
+pub trait ChannelId<T> {
     /// Get the channel.
     fn channel_id() -> ChannelSelection;
 }
 
 macro_rules! impl_channels {
     ($(#[doc = $doc:expr] $CH:ident => [$($IC:ident),+]),+ $(,)?) => {
-        #[derive(Debug, Clone, Copy)]
-        /// ADC input channel selection.
-        pub enum ChannelSelection {
-            $(
-                #[doc = $doc]
-                $CH,
-            )+
+        mod private {
```

Comment:
Why do you move this module inside this macro?

Reply:
This is so the `Sealed` trait can be implemented automatically. Actually, I don't think the trait is even needed anymore.

Reply:
This way the `ChannelId` is sealed by making `ChannelSelection` private.